backend/app/data_processing/gesture.py

```diff
@@ -0,0 +1,101 @@
+import math
+from app.synthesis.synthesizers import generate_sine_wave_with_envelope
+import pyaudio
+
+# factor by which we compress coordinates
+FACTOR = 10
+
+
+def compress_coordinates(gestures):
+    """
+    :param gestures: list of list of coordinates (each coordinate a dictionary) received from user
+    input via
+    frontend
+    :return: (in the same format as input), shorter list of coordinates, each representing a
+    compressed coordinate, compressed by taking averages of groups of coordinates
+    """
+    result = []
+    for gesture in gestures:
+        current_result = []
+        for i in range(0, len(gesture)-FACTOR, FACTOR):
+            compressed_coord = {
+                "x": (gesture[i]["x"] + gesture[i+FACTOR-1]["x"])/2,
+                "y": (gesture[i]["y"] + gesture[i+FACTOR-1]["y"])/2
+            }
+            current_result.append(compressed_coord)
+        result.append(current_result)
+    return result
+
+
+def get_sound(gestures):
+    """
+    Given a list of list of dictionaries {"x": x_val, "y": y_val}, each representing a coordinate
+    from a single gesture received from user input via frontend, return a list of (pitch,
+    duration) tuples representing the gesture converted to sound
+    """
+    result = []
+    for gesture in gestures:
+        for coordinate in gesture:
+            x = coordinate["x"]
+            y = coordinate["y"]
+            pitch = get_pitch(x)
+            duration = get_duration(y)
+            result.append((pitch, duration))
+    return result
+
+
+def get_pitch(x, low=440, high=880):
```

Comment:
This isn't really a fix but a comment: in my own personal development style, I avoid making things arguments until they're actually called in my system with different inputs, rather than predicting that at some point I might need them to be arguments. That is an attempt to keep my systems as simple as possible for as long as possible.

To make that concrete here: `get_pitch` is only ever called with a value for `x`. That means that the default values for `low` and `high` are never changed, so they might as well be constants.

But that has even further knock-on consequences: `number_of_octaves` is set dynamically here, but actually we know for sure that it's always going to be `1`, since it's taking log_2 of 880/440, at which point it actually disappears since really our x_pitch is always set within the range of one octave.

Same kind of thing with `get_duration` below.

This is always a judgement call - perhaps we know for sure we're going to extend the functionality and pass arguments to these functions! - but worth considering when writing a specific system is going to be easier and more robust than writing a general one.